src/DocAssembler/DocAssembler/FileService/Hyperlink.cs

```diff
@@ -0,0 +1,274 @@
+﻿// <copyright file="Hyperlink.cs" company="DocFx Companion Tools">
+// Copyright (c) DocFx Companion Tools. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// </copyright>
+using System.Text;
+
+namespace DocAssembler.FileService;
+
+/// <summary>
+/// Hyperlink in document.
+/// </summary>
+public class Hyperlink
+{
+    private static readonly char[] UriFragmentOrQueryString = new char[] { '#', '?' };
+    private static readonly char[] AdditionalInvalidChars = @"\/?:*".ToArray();
+    private static readonly char[] InvalidPathChars = Path.GetInvalidPathChars().Concat(AdditionalInvalidChars).ToArray();
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Hyperlink"/> class.
+    /// </summary>
+    public Hyperlink()
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Hyperlink"/> class.
+    /// </summary>
+    /// <param name="filePath">Path of the markdown file.</param>
+    /// <param name="line">Line number.</param>
+    /// <param name="col">Column.</param>
+    /// <param name="url">Url.</param>
+    public Hyperlink(string filePath, int line, int col, string url)
+    {
+        FilePath = filePath;
+        Line = line;
+        Column = col;
+
+        Url = url;
+        OriginalUrl = Url;
+
+        LinkType = HyperlinkType.Empty;
+        if (!string.IsNullOrWhiteSpace(url))
+        {
+            if (url.StartsWith("https://", StringComparison.InvariantCulture) || url.StartsWith("http://", StringComparison.InvariantCulture))
```

Comment:
Makes sense if I move this code to a shared library used by the other tool(s) as well. For now, as it is marked as a specific type I suggest to keep it like this for now.

Reply:
you definitely have a repetition of all this here: https://github.com/Ellerbach/docfx-companion-tools/pull/87/files#r1884094489

Why not creating a dictionary with the "https://" and so on as a key and the enum as a value?
It will make it easier to search on the other one as well, as it's just about checking that anything with the key exists and it's easier to maintain

Reply:
Done :)